lib/http/timeout/per_operation.rb

```diff
@@ -59,14 +59,17 @@ def write(data)
         # Read data from the socket
         def readpartial(size)
           loop do
-            result = socket.read_nonblock(size, :exception => false)
+            result = rescue_readable do
```

Comment:
This doesn't make sense, there's nothing to rescue since it's not using exceptions?